private/buf/buffetch/buffetch.go

```diff
@@ -34,6 +34,10 @@ const (
 	ImageEncodingBin ImageEncoding = iota + 1
 	// ImageEncodingJSON is the JSON image encoding.
 	ImageEncodingJSON
+	// MessageEncodingBin is the binary image encoding.
+	MessageEncodingBin MessageEncoding = iota + 1
```

Comment:
I'm worried about this - `buffetch` is meant to be specific to things that can represent modules, namely images, sources, and modules themselves. It seems that `buffetch` is being re-used to do some work with message encoding/decoding. `buffetch/internal` was originally designed to be a library that was independent of what you were trying to fetch, buf `buffetch` is not.

Reply:
do you think it will be better to put these logic to `bufreflect`? we might also need a ref parser to handle the output format, which seems should also live in `bufreflect`

Reply:
I don't think `bufreflect` is the right home either - it's meant to be a lightweight package that interacts with the `protoreflect` APIs. Ideally we just keep it scoped to things like `bufreflect.NewMessage`.

But I agree with @bufdev - this doesn't really make sense in `buffetch` because it should remain specific to modules. We just want similar logic handling the inputs we support for encoding/decoding (e.g. binary protobuf, JSON, etc).

Given that we only have one CLI command, we could start by putting all of this logic in the same package as the command implementation for now. If we ever need it elsewhere, we could create a new home for it and go from there. Let's keep it scoped to just supporting `.bin` and `.json` for now. This wouldn't work with the `bufwire` abstractions in the same way (since it won't be a `buffetch.Ref`), but we might not need all of that to get something tailored to what we want in these commands.

TL;DR Try simplifying some of this logic in the command implementation and see if we can come up with something simpler that doesn't affect `buffetch`.